state/state.go

```diff
@@ -56,3 +56,10 @@ func (a Address) MarshalJSON() ([]byte, error) {
 	// The address is marshaled into a simple string value
 	return []byte("\"" + a.Address.String() + "\""), nil
 }
+
+// PayForBlobOptions contains additional options that will be applied to the `MsgPayForBlob`.
+type PayForBlobOptions struct {
```

Comment:
I would tend to argue that a struct is the wrong approach for this. Once more options are added, it will mean that users must set ALL options

Above I say we could use variadic params, but that doesn't actually work because you cannot pass those types over RPC.

An idea that /might/ work is a map[string]interface{} (I know, ugly, but we can hide it behind a custom type with easy to use constructors and validation methods, possibly using actual custom variadic options). This sounds a bit complex, so we'd have to see it and try it before deciding if it makes sense. Especially because it means this custom type+methods would have to be implemented in the other languages' client libraries to keep good devex.

I don't really like that either, but I think a struct can't work, so we need to find some alternative

Reply:
We could ask for `[]Option` instead and users could pass `DefaultOptions` which is just some empty []opt list.

This is a way to work around variadics not being over RPC

Reply:
The issue isn't directly variadics, it's the actual option type. How is it represented? And how does it get sent over the wire?